crates/db/types/src/traits.rs

```diff
@@ -351,3 +354,39 @@ pub trait L1BroadcastDatabase: Send + Sync + 'static {
     /// Get last broadcast entry
     fn get_last_tx_entry(&self) -> DbResult<Option<L1TxEntry>>;
 }
+
+/// Database trait for toplevel OL state storage.
+///
+/// Stores OLState snapshots keyed by OLBlockCommitment (block ID + slot).
+/// This allows retrieving state for any block in the chain.
+pub trait ToplevelOLStateDatabase: Send + Sync + 'static {
+    /// Stores a toplevel OLState snapshot for a given block commitment.
+    fn put_toplevel_ol_state(&self, commitment: OLBlockCommitment, state: OLState) -> DbResult<()>;
+
+    /// Retrieves a toplevel OLState snapshot for a given block commitment.
+    fn get_toplevel_ol_state(&self, commitment: OLBlockCommitment) -> DbResult<Option<OLState>>;
+
+    /// Gets the latest toplevel OLState (highest slot).
+    fn get_latest_toplevel_ol_state(&self) -> DbResult<Option<(OLBlockCommitment, OLState)>>;
+
+    /// Deletes a toplevel OLState snapshot for a given block commitment.
+    fn del_toplevel_ol_state(&self, commitment: OLBlockCommitment) -> DbResult<()>;
```

Comment:
Trait should be `OLStateDatabase` since we're storing a lot more than just the toplevel state here.